test/regress/regress1/strings/issue5611-deq-norm-emp.smt2

```diff
@@ -7,4 +7,5 @@
 (assert (not (= str8 (str.replace_re_all (str.++ str9 str8 str7) (str.to_re "ULfQhzdcQSfqWSXyjuptqjqsazpyjzdDddlJPLDJhalmhBhlNBKZvoxoLOpfplkqhvIRHMOMDAGIdoRyiZmxmMvRijgpFnd") str9))))
 (push)
 (assert (and (str.in_re str8 (str.to_re str7)) (not (= str9 (str.replace_re_all (str.++ str8 str8) (str.to_re "ULfQhzdcQSfqWSXyjuptqjqsazpyjzdDddlJPLDJhalmhBhlNBKZvoxoLOpfplkqhvIRHMOMDAGIdoRyiZmxmMvRijgpFnd") str9)))))
+; note that (debug) model checking fails since this benchmark triggers a string variable (str7) to be solved after it appears in assertions
```

Comment:
What does this mean?

Reply:
--debug-check-models is an option which checks whether assertions are satisfied after each "sat".

In this benchmark a string variable `str7` is solved during non-clausal simplification after we already push assertions involving `str7` to the solver.  This causes debug-check-models to fail.